R/France.R

```diff
@@ -0,0 +1,126 @@
+    common_data_urls = list(),
+    # nolint start
+    #' @field level_data_urls List of named links to raw data that are level
+    #' specific.
+    level_data_urls = list(
+      "1" = list("cases" = "https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01"),
+      "2" = list(
+        "cases" = "https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675",
+        "hosp" = "https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c"
+      )
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "tested_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    #' @importFrom dplyr select
+    set_region_codes = function() {
+      if (self$level == "1") {
+        self$codes_lookup$`1` <- france_codes %>%
+          select(
+            .data$level_1_region_code,
+            .data$level_1_region,
+            .data$insee_code
+          )
+      } else if (self$level == "2") {
+        self$codes_lookup$`2` <- france_codes
+      }
+    },
+
+    #' @description France Specific Region Level Data Cleaning
+    #'
+    #' @importFrom dplyr filter mutate left_join select
+    #' @importFrom lubridate as_date ymd
+    clean_level_1 = function() {
+      self$data$clean <- self$data$raw$cases %>%
+        filter(cl_age90 == 0) %>%
+        select(
+          date = jour,
+          insee_code = reg,
+          cases_new = P,
+          tested_new = `T`
+        ) %>%
+        mutate(date = as_date(ymd(date))) %>%
+        left_join(
+          self$codes_lookup$`1`,
+          by = c("insee_code")
+        )
+    },
+
+    #' @description France Specific Department Level Data Cleaning
+    #'
+    #' @importFrom dplyr filter mutate left_join rename select full_join
+    #' @importFrom lubridate as_date ymd
+    clean_level_2 = function() {
+      cases_data <- self$data$raw$cases %>%
+        filter(cl_age90 == 0) %>%
+        select(
+          date = jour,
+          level_2_region_code = dep,
+          cases_new = P,
+          tested_new = `T`
+        ) %>%
+        mutate(date = as_date(ymd(date)))
+
+      if (!is.null(self$data$raw$hosp)) {
+        hosp_data <- self$data$raw$hosp %>%
+          select(
+            date = jour,
+            level_2_region_code = dep,
+            hosp_new = incid_hosp,
+            deaths_new = incid_dc
+          ) %>%
+          mutate(date = as_date(ymd(date)))
+
+        combined_data <- full_join(cases_data, hosp_data,
+          by = c("date", "level_2_region_code")
+        )
+      } else {
+        combined_data <- cases_data
+      }
+
+      self$data$clean <- combined_data %>%
+        mutate(
+          level_2_region_code =
+            paste0("FR-", level_2_region_code)
+        ) %>%
+        left_join(self$codes_lookup$`2`, by = "level_2_region_code")
```

Comment:
if possible joining in level 1 codes would be neat. If not no problem

Reply:
level 1 codes are included in `self$codes_lookup[["2"]]`